content/posts/networkx/vf2pp/ISO-feasibility-candidates/index.md

```diff
@@ -0,0 +1,156 @@
+---
+title: "ISO Feasibility & Candidates"
+date: 2022-07-11
+draft: false
+description: "
+Information about my progress on two important features of the algorithm.
+"
+tags: ["gsoc", "networkx"]
+displayInList: true
+author: ["Konstantinos Petridis"]
+
+---
+
+The previous post can be found [here]({{< relref "../node-ordering-Ti-updating" >}}), be sure to check it out, so you
+can
+follow the process step by step. Since then, another two very significant features of the algorithm have been
+implemented and tested.
+
+## Introduction
+
+As it was previously mentioned, in the ISO problem we are basically trying to create a **mapping** such that, every node
+from the first graph is matched to a node from the second graph. This searching for "feasible pairs" can be visualized
+by a tree, where each node is the candidate pair that we should examine. This can become much clearer if we take a look
+at the below figure.
+
+<center><img src="dfs.png" alt="DFS VF2++ example$"/></center>
+
+In order to check if the graphs $G_1$, $G_2$ are isomorphic, we check every candidate pair of nodes and if it is
+feasible, we extend the mapping and go deeper into the tree of pairs. If it's not feasible, we climb up and follow a
+different branch, until the mapping is extended to its full. In our example, we start by examining node 0 from G1, with
+node 0 of G2. After some checks that are skipped for simplicity (but can be perceived by observation), we decide that the
+nodes 0 and 0 are matching, so we go deeper to map the remaining nodes. The next pair is 1-1, which fails the
```

Comment:
This part is actually a bit confusing because there is no node `1` in G_2. Of course - this is the whole point: it's not feasible to map node `1` in G_1 to node `1` in G_2 because node `1` in G_2 doesn't exist.

In fact, is this a good example? AIUI from reading on, node `1` in G_2 actually *wouldn't* be part of the candidate nodes in G_2 because the first prerequisite for creating the candidate set is that the nodes must exist in G_2, right?

Perhaps the example could be updated to use `1-3` as the next candidate, which would not be feasible because of the degree mismatch.

Unfortunately I think this may also necessitate updating the left-most figure in the accompanying image, as I don't think checking `1-1` makes sense --- unless I'm completely misunderstanding candidate selection!

Reply:
No you are right, i was too lazy to check. Your understanding is fine. Basically we would try to match nodes 1-1 if 1 was present in G2 (which i didn't notice it wasn't) so i will just relabel the nodes in G2 or just pick another candidate for 1. I wanted to shift the emphasis on the logic of the procedure rather than the details, but this could indeed be misleading. I will correct when immediately. Thanks for noticing!

Reply:
All done!